library/Businessprocess/MonitoringRestrictions.php

```diff
@@ -62,4 +62,32 @@ function ($c) {
 
         return $restriction;
     }
+
+    public function getCustomFilterWithRestriction($restrictionName, $filterQueryString) {
```

Comment:
I don't quite understand the reason for this method.

It limits the available columns to filter for, and that's the only visible intention. But why shouldn't one be able to use any filter here? After all, it should work to place a filter here that's been set up in the host/service listing using the filter editor.

Reply:
You're right. I updated the function in order to accept any filter and then apply user monitoring restrictions.

Reply:
Yeah, well, but now it doesn't have anything to do with restrictions anymore. I don't see a reason for a method here anyway.

Add the filters directly to the queries instead. e.g.

```php
$query->addFilter($this->getRestriction('monitoring/filter/objects'))
    ->addFilter(Filter::fromQueryString($filterQueryString))
```